security/secret-management.md

```diff
@@ -0,0 +1,108 @@
+  secrets are rarely shared with developers and they tend to be used only in production and staging environments.
+- **Test Secret**: A secret used in non-production environments, such as development or testing, used for testing 
+  purposes only. However, Test secrets need to be carefully handled as they might grant access to services or result in
+  a breach of agreement with third-party systems. 
+- **Non-Secret Key**: An API key that does not provide access to sensitive data or critical functionalities and is 
+  considered low-risk if exposed. The most common example is a key that needs to be used by JavaScript and can be read 
+  in the source code of a page. While it is not classified as a secret, it is still preferred to manage it properly to 
+  maintain best practices in code management and security hygiene. See more information about 
+  [Non-Secret Key Management](key-management.md)
+
+### Types of Secrets
+
+- **API Keys**: Credentials used to authenticate applications with external services.
+- **Passwords**: Credentials used to access databases, services, or systems.
+- **Tokens**: Access tokens or refresh tokens used for authentication and authorization.
+- **Certificates**: SSL/TLS certificates or other cryptographic keys used for securing communications.
+- **Environment Variables**: Sensitive data stored as environment variables in deployment configurations.
+
+
+## What to do when a vulnerability is found
+
+### Assessment
+
+1. **Alert**: Use GitGuardian to detect and alert on leaked secrets on code commit.  
+2. **Review**: If the secret is a false positive. In case of false positive resolve the issue and including an 
+   explanation why the issue is considered a false positive. Note: Test Secrets are not false positive and they need to
+   be carefully considered as they might grant access to public facing services
+3. **Determine the Type**: Identify whether the leaked secret is a production secret, a test secret or a non-secret key.
+4. **Assess the Impact**: Evaluate the severity and potential impact of the leak on systems, data, and user security.
+5. **Open Source**: Identify whether the project is Open Source or not. Open Source projects
+6. **Report**: Report the issue to your technical leader to make sure that the issue has been dealt correctly. They 
+   might help to validate that the proper action has been or will be performed
+   
+### Remediation
+If the secret relates to an Open Source Repository
+1. **Revoke the Secret**: Immediately revoke the leaked secret to prevent unauthorized access. For instance, rotate API 
+   keys, reset passwords, or invalidate tokens.
+2. **Update Configuration**: Replace the leaked secret with a new, secure value in all relevant systems and codebases.
+3. **Remove from History**: Ensure the secret is removed from the code repository history. Use tools like 
+ _git filter-branch_ or _BFG Repo-Cleaner_ to clean the repository history.
+
```

Comment:
Can we link to any documentation here?

Reply:
Good Shout!

Reply:
I have added a link to GitHub recomendations - https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository